### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by kafkaphoenix**
Hi I think you should move civilization and world to their own classes using a "manager" class to store the instances and execute the actions you are currently doing in the main method, that way it would be easier to modularize the code and improve readability. I can help with that if you are interested. Cheers

### CODE REVIEW

**Comment by kafkaphoenix** [comment]
I would suggest to move World and Civilisation to their own classes and then create a third class "manager" which has the reference to the instances and execute the actions that you are doing now in the main method. Only one file is harder to maintain. I can help with that if you want to do it.

**Comment by kafkaphoenix** [comment]
Added an utils file with the auxiliar methods
Reordered imports
Replaced for variables with _ if they aren't used

File: constants.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    88888 '> `8888>      888E    8888  8888  %8"    R88   888E    X888  888X '888>  
+    `8888L %  ?888   !   888E   .8888b.888P   @8Wou 9%    888E    X888  888X '888>  
+     `8888  `-*""   /    888&    ^Y8888*""  .888888P`     888&   "*88%""*88" '888!` 
+       "888.      :"     R888"     `Y"      `   ^"F       R888"    `~    "    `"`   
+         `""***~"`        ""                               ""                       
+    '''
+
+# Biome states
+
+UNOCUPPIED = ''
```
Comment (UtilityHotbar) [inline constants.py]:
```suggestion
UNOCCUPIED = ''
```

File: main.py [hunk 2]
```diff
@@ -1,55 +1,30 @@
...
-HEIGHT = 20
-WIDTH = 50
-AGE = 500  # In years
 TEMP = [[messy_noise([i / WIDTH, j / HEIGHT], tnoise)+0.1 for i in range(WIDTH)] for j in range(HEIGHT)]
 ELEV = [[messy_noise([i / WIDTH, j / HEIGHT], enoise) for i in range(WIDTH)] for j in range(HEIGHT)]
-WORLD = [['?' for i in range(WIDTH)] for j in range(HEIGHT)]
-CIV = [['' for i in range(WIDTH)] for j in range(HEIGHT)]
-CIVLETTERS = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ!@#$%^&*()-_=+[{]}\|;:,<.>/?'
+WORLD = [[OCEAN]*WIDTH for _ in range(HEIGHT)]
+CIV = [[UNOCUPPIED]*WIDTH for _ in range(HEIGHT)]
```
Comment (UtilityHotbar) [inline main.py]:
```suggestion
CIV = [[UNOCCUPIED]*WIDTH for _ in range(HEIGHT)]
```

File: main.py [hunk 3]
```diff
@@ -69,51 +44,54 @@ def __init__(self, name, symbol, origin, ancestor=None, control=False):
...
+            ay = random.randint(clamp(attempt[0]-rng, nmax=HEIGHT-1), clamp(attempt[0]+rng, nmax=HEIGHT-1))
             attempt_loc = WORLD[ay][ax]
-            if not (attempt_loc == '~' or attempt_loc == self.symbol):  # Cannot expand into water
+            
+            if not (attempt_loc == OCEAN or [ay, ax] in self.territory):  # Cannot expand into water
                 c = 10 + max(50, 5*self.techlevel)  # Base 10% success chance, max +50% from technology
                 if attempt_loc == self.home_biome:  # Bonus chance for expansion if attempting to expand into base biome
                     c += 20
-                if CIV[ay][ax] != '':  # Spreading into occupied squares is harder
+                if CIV[ay][ax] != UNOCUPPIED:  # Spreading into occupied squares is harder
```
Comment (UtilityHotbar) [inline main.py]:
```suggestion
                if CIV[ay][ax] != UNOCCUPIED:  # Spreading into occupied squares is harder
```

File: main.py [hunk 4]
```diff
@@ -141,11 +119,11 @@ def collapse(self, decimate=False):
         self.instability += rpgtools.roll(f'{math.ceil(self.powerbase/10)}d6')
         self.power /= random.randint(2, 4)
         seeding_targets = []
-        for x in range(random.randint(math.floor(len(self.territory)/1.5), len(self.territory))):
+        for _ in range(random.randint(math.floor(len(self.territory)/1.5), len(self.territory))):
             rmtarget = random.choice(self.territory)
             seeding_targets.append(rmtarget)
             self.territory.remove(rmtarget)
-            CIV[rmtarget[1]][rmtarget[0]] = ''
+            CIV[rmtarget[0]][rmtarget[1]] = UNOCUPPIED
```
Comment (UtilityHotbar) [inline main.py]:
```suggestion
            CIV[rmtarget[0]][rmtarget[1]] = UNOCCUPIED
```

File: utils.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+def scan(world, target, delete=False):
+    reslist = []
+    for y in range(HEIGHT):
+        for x in range(WIDTH):
+            if world[y][x] == target:
+                if not delete:
+                    reslist.append([y, x])
+                else:
+                    world[y][x] = UNOCUPPIED
```
Comment (UtilityHotbar) [inline utils.py]:
```suggestion
                    world[y][x] = UNOCCUPIED
```

File: utils.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
+from constants import BEACH, DESERT, FOREST, GRASSLAND, HEIGHT, JUNGLE, MOUNTAIN, OCEAN, SAVE_LOCATION, SNOW, WIDTH, UNOCUPPIED
```
Comment (UtilityHotbar) [inline utils.py]:
```suggestion
from constants import BEACH, DESERT, FOREST, GRASSLAND, HEIGHT, JUNGLE, MOUNTAIN, OCEAN, SAVE_LOCATION, SNOW, WIDTH, UNOCCUPIED
```

**Review by UtilityHotbar — APPROVED** [review]

**Comment by kafkaphoenix** [comment]
thanks for fixing the typo, @UtilityHotbar feel free to merge it

**Pull request merged**